client/alloc_watcher.go

```diff
@@ -517,9 +518,14 @@ func (p *remotePrevAlloc) streamAllocDir(ctx context.Context, resp io.ReadCloser
 				f.Close()
 				return fmt.Errorf("error chmoding file %v", err)
 			}
-			if err := f.Chown(hdr.Uid, hdr.Gid); err != nil {
-				f.Close()
-				return fmt.Errorf("error chowning file %v", err)
+
+			// Can't change owner if not root. Returns false on
+			// Windows as Chown always errors there.
+			if syscall.Geteuid() == 0 {
```

Comment:
You should cache this as you will potentially be making a lot of syscalls!

Reply:
Good call. Fixed!